src/phoenix/datasets/tracing.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import annotations
+
+from binascii import hexlify
+from contextlib import contextmanager
+from contextvars import ContextVar
+from threading import Lock
+from typing import (
+    Any,
+    Callable,
+    Iterator,
+    Optional,
+)
+
+from openinference.semconv.resource import ResourceAttributes
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.trace import ReadableSpan
+from opentelemetry.sdk.trace.id_generator import RandomIdGenerator
+from opentelemetry.trace import INVALID_TRACE_ID, SpanContext
+from wrapt import apply_patch, resolve_path, wrap_function_wrapper
+
+from phoenix.datasets.types import TraceId
+
+
+class TraceStealer:
+    __slots__ = ("_trace_id", "_trace_found", "_resource")
+
+    def __init__(self, project_name: Optional[str] = None) -> None:
+        self._trace_id: int = _ID_GEN.generate_trace_id()
+        self._trace_found = False
+        self._resource = (
+            Resource.create({ResourceAttributes.PROJECT_NAME: project_name})
+            if project_name
+            else None
+        )
+
+    @property
+    def trace_id(self) -> Optional[TraceId]:
+        if self._trace_found:
+            return _str_trace_id(self._trace_id)
+        return None
+
+    def mutate(self, span: ReadableSpan) -> None:
```

Comment:
can you add some documentation here? mutate is a bit too generic and it's hard to understand what it's doing. Maybe something like `inject_trace_context` or something like that might be more descriptive?